requirements_dev.txt

```
@@ -18,7 +18,7 @@ nbclient
 scikit-learn
 tensorflow>=1.15.2; sys_platform != 'darwin'
 tensorflow>=1.15.2; python_version > '3.6' and sys_platform == 'darwin' and platform.machine != 'arm64'
-tensorflow-macos; python_version > '3.6' and python_version < '3.10' and sys_platform == 'darwin' and platform.machine == 'arm64'
+tensorflow-macos; sys_platform == 'darwin' and platform.machine == 'arm64'
```

We haven't dropped support for Python 3.6 yet; does `tensorflow-macos` install on with Python 3.6? If not we should keep the lower bound.

---

I checked, and the answer is no. The tests that need `tensorflow` won't pass on Python 3.6/macos/M1, but if we don't put the lower bound back the install will fail and prevent the other tests from running.
```suggestion
tensorflow-macos; python_version > '3.6' and sys_platform == 'darwin' and platform.machine == 'arm64'
```

---

The same is true for Python 3.7 though: there's no wheel for py37, so installation will fail. Should we do this then?
```suggestion
tensorflow-macos; python_version > '3.7' and sys_platform == 'darwin' and platform.machine == 'arm64'
```

---

For reference, I see CI only runs tests on MacOS with Python 3.9.